pytrip/utils/dddplot.py

```diff
@@ -0,0 +1,169 @@
+#    GNU General Public License for more details.
+#
+#    You should have received a copy of the GNU General Public License
+#    along with PyTRiP98.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+Script for generating DVH or other volume histograms.
+"""
+import os
+import sys
+import logging
+import argparse
+import numpy as np
+import pytrip as pt
+
+logger = logging.getLogger(__name__)
+
+
+class DDD(object):
+    """This class handles DDD files."""
+    # TODO: this class could go into main pytrip/ as well.
+
+    def __init__(self):
+        """ setup of object """
+        self.path = None
+        self.filename = None  # filename, without directory
+        self.filetype = "ddd"
+        self.fileversion = "19980520"  # default
+        self.filedate = None  # format like "Sat Sep 12 14:48:56 2020"
+        self.projectile = ""  # "12C"
+        self.material = ""  # H2O
+        self.composition = ""  # H2O
+        self.density = 1.0
+        self.energy = 0.0  # in MeV/u (NOT per nucleon)
+        self.ngauss = 0  # number of gaussian fits found in this file
+        self.ndata = 0  # number of data points in this set
+        self.data = None
+
+    def read(self, fn):
+        """ Reads the filename set when constructing the ReadGd class
+        """
+        if os.path.isfile(fn) is False:
+            raise IOError("Could not find file " + fn)
+
+        self.path = fn
+        self.filename = os.path.basename(fn)
+
+        with open(fn) as f:
+            lines = f.readlines()
+
+        for i, line in enumerate(lines):
+            _l = line.strip().split(' ', 1)
+            if len(_l) == 2:
+                key, value = _l
+            if key == "!filetype":
+                self.filetype = value
+            if key == "!fileversion":
+                self.fileversion = value
+            if key == "!filedate":
+                self.filedate = value
+            if key == "!projectile":
+                self.projectile = value
+            if key == "!material":
+                self.material = value
+            if key == "!composition":
+                self.composition = value
+            if key == "!density":
+                self.density = float(value)
+            if key == "!energy":
+                self.energy = float(value)
+
+        self.data = np.loadtxt(fn, comments=("#", "!"))
+
+        rows, cols = self.data.shape
+
+        if rows < 1:
+            raise IOError("Invalid number of data points in DDD file: " + fn)
+        self.ndata = rows
+
+        if cols < 2:
+            raise IOError("Invalid number of columns in DDD file: " + fn)
+        self.ngauss = cols - 2  # number of gaussian fits donw
+
+    def savefig(self, type="png"):
+        """
+        """
+        import matplotlib
+        matplotlib.use('Agg')
+        import matplotlib.pyplot as plt
+
+        # add suffix
+        outfile = self.filename + "." + type
+
+        fig, (p1, p2) = plt.subplots(2, 1)
```

Comment:
What do you think of setting bigger filesize ? I.e. (15,10) ?

Reply:
?

Reply:
I mean a figsize, not filesize, i.e:  `fig, ax = plt.subplots(2,1, figsize=(20,10))`